src/Nethermind/Nethermind.Merge.AuRa/AuRaMergeBlockProducerEnvFactory.cs

```diff
@@ -0,0 +1,160 @@
+using Nethermind.Consensus.AuRa.Transactions;
+using Nethermind.Consensus.Comparers;
+using Nethermind.Consensus.Processing;
+using Nethermind.Consensus.Producers;
+using Nethermind.Consensus.Rewards;
+using Nethermind.Consensus.Transactions;
+using Nethermind.Consensus.Validators;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Db;
+using Nethermind.Logging;
+using Nethermind.State;
+using Nethermind.Trie.Pruning;
+using Nethermind.TxPool;
+
+namespace Nethermind.Merge.AuRa
+{
+    public class AuRaMergeBlockProducerEnvFactory : BlockProducerEnvFactory
+    {
+        private readonly AuRaNethermindApi _auraApi;
+        private readonly IAuraConfig _auraConfig;
+        private readonly DisposableStack _disposeStack;
+
+        public AuRaMergeBlockProducerEnvFactory(
+            AuRaNethermindApi auraApi,
+            IAuraConfig auraConfig,
+            DisposableStack disposeStack,
+            IDbProvider dbProvider,
+            IBlockTree blockTree,
+            IReadOnlyTrieStore readOnlyTrieStore,
+            ISpecProvider specProvider,
+            IBlockValidator blockValidator,
+            IRewardCalculatorSource rewardCalculatorSource,
+            IReceiptStorage receiptStorage,
+            IBlockPreprocessorStep blockPreprocessorStep,
+            ITxPool txPool,
+            ITransactionComparerProvider transactionComparerProvider,
+            IMiningConfig miningConfig,
+            ILogManager logManager) : base(
+                dbProvider,
+                blockTree,
+                readOnlyTrieStore,
+                specProvider,
+                blockValidator,
+                rewardCalculatorSource,
+                receiptStorage,
+                blockPreprocessorStep,
+                txPool,
+                transactionComparerProvider,
+                miningConfig,
+                logManager)
+        {
+            _auraApi = auraApi;
+            _auraConfig = auraConfig;
+            _disposeStack = disposeStack;
+        }
+
+        protected override BlockProcessor CreateBlockProcessor(
+            ReadOnlyTxProcessingEnv readOnlyTxProcessingEnv,
+            ISpecProvider specProvider,
+            IBlockValidator blockValidator,
+            IRewardCalculatorSource rewardCalculatorSource,
+            IReceiptStorage receiptStorage,
+            ILogManager logManager,
+            IMiningConfig miningConfig)
+        {
+            return new AuRaMergeBlockProcessor(
+                specProvider,
+                blockValidator,
+                rewardCalculatorSource.Get(readOnlyTxProcessingEnv.TransactionProcessor),
+                TransactionsExecutorFactory.Create(readOnlyTxProcessingEnv),
+                readOnlyTxProcessingEnv.StateProvider,
+                readOnlyTxProcessingEnv.StorageProvider,
+                receiptStorage,
+                logManager,
+                _blockTree);
+        }
+
+        protected override TxPoolTxSource CreateTxPoolTxSource(
```

Comment:
Seems redundant if we already kept code in one place thanks to below. While we are exposing some internal AuRa stuff to be used here it is fine.

```
return new StartBlockProducerAuRa(_auraApi!)
    .CreateStandardTxSourceForProducer(txProcessingEnv, constantContractsProcessingEnv);
```

Reply:
done